### PR TITLE
Configure environment variables in workloads

### LINKED ISSUE #688: Add API config to pass environment variables through to containers

### PR DESCRIPTION
**Pull request by vishalbollu**
closes #688 

---

checklist:

- [ ] run `make test` and `make lint`
- [ ] test end to end manually (i.e. build/push all images, restart operator, and run `cortex deploy --refresh`)
- [ ] update the examples
- [ ] update the docs and add any new files to `summary.md`
- [ ] merge to master
- [ ] cherry-pick into release branches if applicable
- [ ] check that the docs are up-to-date in [gitbook](https://www.cortex.dev/v/master)
- [ ] alert the dev team if the dev environment changed
- [ ] delete the branch once it's merged

### CODE REVIEW

File: pkg/operator/api/userconfig/apis.go [hunk 1]
```diff
@@ -59,6 +59,13 @@ var configValidation = &cr.StructFieldValidation{
 	},
 }
 
+var envValidation = &cr.StructFieldValidation{
+	StructField: "Env",
+	StringMapValidation: &cr.StringMapValidation{
+		Default: map[string]string{},
```
Comment (deliahu) [inline pkg/operator/api/userconfig/apis.go]: Do we need `AllowEmpty: true`?

File: pkg/operator/workloads/api_workload.go [hunk 2]
```diff
@@ -299,16 +299,26 @@ func tfAPISpec(
 		},
```
Comment (deliahu) [inline pkg/operator/workloads/api_workload.go]: Do we need to update the ID calculation in `operator/context/apis.go`? While you have that file open, do you mind checking if we missed anything else we should add there?

File: pkg/operator/workloads/api_workload.go [hunk 3]
```diff
@@ -484,16 +495,26 @@ func pythonAPISpec(
 	downloadArgsBytes, _ := json.Marshal(downloadConfig)
 	downloadArgsStr := base64.URLEncoding.EncodeToString(downloadArgsBytes)
 
-	envVars := []kcore.EnvVar{
-		{
+	envVars := []kcore.EnvVar{}
+
+	for name, val := range api.TensorFlow.Env {
```
Comment (deliahu) [inline pkg/operator/workloads/api_workload.go]: should this be `api.Python.Env`?

File: pkg/operator/workloads/api_workload.go [hunk 4]
```diff
@@ -634,16 +655,26 @@ func onnxAPISpec(
 		},
 	}
 
-	envVars := []kcore.EnvVar{
-		{
+	envVars := []kcore.EnvVar{}
+
+	for name, val := range api.TensorFlow.Env {
```
Comment (deliahu) [inline pkg/operator/workloads/api_workload.go]: should this be `api.ONNX.Env`?

File: pkg/operator/workloads/api_workload.go (hunk 2 above)
Comment (vishalbollu) [reply pkg/operator/workloads/api_workload.go]: checked

**Pull request merged**